### PR TITLE
Enable postgres for installing hub-of-hubs

### PR DESCRIPTION
**Pull request by yanmxa**
fixed: https://github.com/stolostron/backlog/issues/23504
Signed-off-by: Meng <yanmxa@163.com>

### CODE REVIEW

**Comment by yanmxa** [comment]
https://github.com/stolostron/backlog/issues/23504

File: test/setup/postgres/postgres_setup.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (clyang82) [inline test/setup/postgres/postgres_setup.sh:1]: You missed `#!/bin/bash`

File: test/setup/postgres/postgres_setup.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+function deployHohPostgresqlAction() {
+  cd ./pgo
+  IMAGE=quay.io/open-cluster-management-hub-of-hubs/postgresql-ansible:$TAG ./setup.sh
+  cd ..
+}
+
+function deployPostgres() {
+  # always check whether DATABASE_URL_HOH and DATABASE_URL_TRANSPORT are set, if not - install PGO and use its secrets
+  if [ -z "${DATABASE_URL_HOH-}" ] && [ -z "${DATABASE_URL_TRANSPORT-}" ]; then
+    deployComponent "hub-of-hubs-postgresql" "$branch" deployHohPostgresqlAction
```
Comment (clyang82) [inline test/setup/postgres/postgres_setup.sh:27]: Do you need to define `branch` or it is passed from parent script?

File: test/setup/postgres/postgres_setup.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  # action
+  $3
+  # after
+  cd ..
+  rm -rf $1
+}
+
+function deployHohPostgresqlAction() {
+  cd ./pgo
+  IMAGE=quay.io/open-cluster-management-hub-of-hubs/postgresql-ansible:$TAG ./setup.sh
```
Comment (clyang82) [inline test/setup/postgres/postgres_setup.sh:20]: That is OK for now. in future, we need to separate the postgres installation and DB init.

**Review by clyang82 — CHANGES_REQUESTED** [review]
This is not a postgres setup. this is

File: test/setup/postgres/postgres_setup.sh (hunk 2 above)
Comment (clyang82) [reply test/setup/postgres/postgres_setup.sh:27]: we may need to have a default value for it.

File: test/setup/postgres/postgres_setup.sh (hunk 3 above)
Comment (clyang82) [reply test/setup/postgres/postgres_setup.sh:20]: maybe we move `https://github.com/stolostron/hub-of-hubs-postgresql/tree/main/pgo/install` here. we do not need to enable HA for tests. WDYT?

File: test/setup/postgres/postgres_setup.sh (hunk 2 above)
Comment (yanmxa) [reply test/setup/postgres/postgres_setup.sh:27]: > Do you need to define `branch` or it is passed from the parent script?

It is passed from the parent script. It's better to set a default value for the branch.

File: test/setup/postgres/postgres_setup.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    while [ -z "$matched" ]; do
+      echo "Waiting for secret $secret to be created in namespace $namespace"
+      matched=$(kubectl get secret $secret -n $namespace --ignore-not-found=true)
+      sleep 10
+    done
+  done
+
+  kubectl delete -f ${currentDir}/postgres-job.yaml --ignore-not-found=true
+
+  IMAGE=quay.io/open-cluster-management-hub-of-hubs/postgresql-ansible:$TAG
```
Comment (clyang82) [inline test/setup/postgres/postgres_setup.sh:39]: Can you move the init job out of here? It does not belong to postgres setup. it belongs to hoh installation.

**Pull request merged**